pkg/storj/segmentid.go

```diff
@@ -0,0 +1,84 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package storj
+
+import (
+	"encoding/base32"
+
+	"github.com/zeebo/errs"
+)
+
+// ErrSegmentID is used when something goes wrong with a segment ID
+var ErrSegmentID = errs.Class("segment ID error")
+
+// segmentIDEncoding is base32 without padding
+var segmentIDEncoding = base32.StdEncoding.WithPadding(base32.NoPadding)
+
+// SegmentID is the unique identifier for segment related to object
+type SegmentID []byte
+
+// SegmentIDFromString decodes an base32 encoded
+func SegmentIDFromString(s string) (SegmentID, error) {
+	idBytes, err := segmentIDEncoding.DecodeString(s)
+	if err != nil {
+		return SegmentID{}, ErrSegmentID.Wrap(err)
+	}
+	return SegmentIDFromBytes(idBytes)
+}
+
+// SegmentIDFromBytes converts a byte slice into a segment ID
+func SegmentIDFromBytes(b []byte) (SegmentID, error) {
+	id := make([]byte, len(b))
+	copy(id, b)
+	return id, nil
```

Comment:
is there any reason to have the signature return an error ?

Reply:
this is a temporary version of method to support Metainfo refactoring transition. In future this method will be checking is `b` length is correct. I can remove `error` for now if you think its important.

Reply:
maybe just a comment to explain but I'm fine leaving it in if it was intentional

Reply:
comment added